04-make_epochs.py

```diff
@@ -69,6 +71,12 @@ def run_epochs(subject, session=None):
     raw = mne.concatenate_raws(raw_list)
 
     events, event_id = mne.events_from_annotations(raw)
+
+    if len(np.unique(events[:, 0])) != len(events):
+        warnings.warn('Events are unique. Uniqifying them')
+        _, idx = np.unique(events[:, 0], return_index=True)
+        events = events[idx]
```

Comment:
where does it crash otherwise?

Reply:
It crashes when you try to construct epochs.

Reply:
why do we have 2 events at the same time on this dataset? can we provide the event_ids/conditions we need in config to avoid this hack?

Reply:
@sappelhoff can you dig into it? Thank you!

Reply:
I think it could be because the same trial_type can have different trigger values. See the `trial_type` and `value` column in the excerpt from the `_events.tsv` file from this data.

onset | duration | trial_type | response_time | sample | value
-- | -- | -- | -- | -- | --
21.236328125 | n/a | UncuedWalking | n/a | 10873 | 100
21.84765625 | n/a | UncuedWalking | n/a | 11186 | 200
22.447265625 | n/a | UncuedWalking | n/a | 11493 | 100
23.05078125 | n/a | UncuedWalking | n/a | 11802 | 200
23.642578125 | n/a | PreferredCadence | n/a | 12105 | 114
23.9140625 | n/a | PreferredCadence | n/a | 12244 | 444
24.251953125 | n/a | PreferredCadence | n/a | 12417 | 201
24.47265625 | n/a | PreferredCadence | n/a | 12530 | 251

Reply:
how does that explain the first column of events array not being unique?

Reply:
yes, scratch my comment ... I guess the issue is rather that MNE cannot deal with two events happening at the same time. For sub-01, run-01, there are 5 events that happen at the same time.

I guess this could be due to the fact that this was a MOBI study with lots of different sensors and hardware interfacing?

Reply:
did you find duplicate entries in the first column though? Just to be sure it's not a bug in `mne-bids` ...

Reply:
yes, I was reading it with `mne.io.read_raw_eeglab`

Reply:
I mean, were there duplicate entries in `_events.tsv` that you could find just by Ctrl + F?

Reply:
Yes, there were ... the following:
```
105.6484375
108.759765625
184.498046875
487.634765625
667.81640625
```

but these are also duplicates in the raw `.set` data :-) ... note: this data was not generated with MNE-BIDS ... it's just some BIDS data from openneuro

Reply:
I think this is an idiosyncrasy of the dataset ... **perhaps** we could do something about this in MNE-Python (... "allow" events to be at the same time?) - but we could also ignore it :-)

Reply:
I have hit this problem before and it's annoying. It would be good to deal with this at the MNE end. Maybe the solution is `event_repeated='error'` (current behavior I think?) or `event_repeated='merge'` which will simply do `'%s/%s' % (event_1_key, event_2_key)`, or something similar. Feel free to open an MNE issue about this so that we can discuss.

Reply:
done, https://github.com/mne-tools/mne-python/issues/6665

@jasmainak what's the plan with this PR? wait until issue on MNE-Python is solved? Or merge it?

Reply:
I would wait.

Now that you're in your last week, we should try to simplify the scripts and remove any cruft as much as possible.

Reply:
yep, I am getting in clean-up-mode already ... on my list:

- openneuro fetcher in MNE-Python
- 0.3 release MNE-BIDS
- handling EEG coords/elecs in MNE-BIDS
- THIS PR (https://github.com/mne-tools/mne-study-template/pull/41)
- salvaging as much as possible from https://github.com/mne-tools/mne-bids/pull/244 and then closing to wait for the next attempt

Reply:
maybe: https://github.com/mne-tools/mne-study-template/issues/47 if you have time? Might be easy to tackle if you adapt from [here](https://github.com/mne-tools/mne-biomag-group-demo/blob/master/scripts/processing/99-make_reports.py). Then add it in CircleCI